cpp/mrc/src/internal/utils/contains.hpp

```diff
@@ -29,13 +29,15 @@ bool contains(const ContainerT& container, const KeyT& key)
 }
 
 template <typename C>
-class KeyIterator : public std::iterator<std::bidirectional_iterator_tag,
-                                         typename C::key_type,
-                                         typename C::difference_type,
-                                         typename C::pointer,
-                                         typename C::reference>
+class KeyIterator
 {
   public:
+    using iterator_category = std::bidirectional_iterator_tag;
+    using value_type        = C::key_type;
+    using difference_type   = C::difference_type;
+    using pointer           = C::pointer;
+    using reference         = C::reference;
+
```

Comment:
Is this change unrelated to the upgrade?

Reply:
This wouldn't compile without the change. I think it's related to the gxx version update.